src/WebJobs.Script.WebHost/Routing/ScriptRouteHandler.cs

```diff
@@ -71,7 +71,7 @@ public ScriptRouteHandler(ILoggerFactory loggerFactory, ScriptHost scriptHost)
             bool authorized = await AuthenticateAndAuthorizeAsync(context, descriptor);
             if (!authorized)
             {
-                return new UnauthorizedResult();
+                return descriptor.Metadata.IsDisabled ? (StatusCodeResult)new NotFoundResult() : new UnauthorizedResult();
```

Comment:
>descriptor.Metadata.IsDisabled [](start = 23, length = 30)

It seems odd that we'd need a Disable check specifically for the not authorized path.  Is there a more common place the check can occur that catches both Authorized and Unathorized.

Reply:
This is not really part of the authorization, though.

The logic is: If this wasn't authorized, but the function was disabled, we return 404 instead of a 401

@alrod I know when we discussed this we talked about using the ternary conditional operator, but I think this would look a bit cleaner/easier to read like: 

``` csharp
            bool authorized = await AuthenticateAndAuthorizeAsync(context, descriptor);
            if (!authorized)
            {
                if (descriptor.Metadata.IsDisabled)
                {
                     return new NotFoundResult();
                }
                return  new UnauthorizedResult();
            }
```